packages/rules/src/bannedTransitiveDependencies.ts

```diff
@@ -0,0 +1,40 @@
+/*!
+ * Copyright 2020 Palantir Technologies, Inc.
+ *
+ * Licensed under the MIT license. See LICENSE file in the project root for details.
+ *
+ */
+
+import { Context, RuleModule } from "@monorepolint/core";
+import { LockFileObject, parse as lockfileParse } from "@yarnpkg/lockfile";
+import { readFileSync } from "fs";
+import path from "path";
+import * as r from "runtypes";
+
+const Options = r.Record({
+  bannedTransitiveDependencies: r.Array(r.String),
+});
+
+type Options = r.Static<typeof Options>;
+
+export const bannedTransitiveDependencies: RuleModule<typeof Options> = {
+  check: (context: Context, opts: Options) => {
+    const { bannedTransitiveDependencies: bannedDeps } = opts;
+
+    const packagePath = context.getWorkspaceContext().packageDir;
+    const yarnLockFilePath = path.join(packagePath, "yarn.lock");
+
+    const yarnLock: LockFileObject = lockfileParse(readFileSync(yarnLockFilePath).toString()).object;
+    const allDeps = Object.keys(yarnLock);
```

Comment:
we only care about direct deps though right? e.g. if i pull in a banned dep as a dev dep for types only, do we care?

Reply:
I suppose so, but parsing yarn.lock its not possible to know what is and isn't dev dependencies right? We'd have to do a dependency tree parse for that (I naively tried a `npm ls X` solution but it was way too slow). Might need to pull some of Walker's DependencyTreeService out.

Reply:
hm looking at yarn.lock i don't think transitive dev deps show up in yarn.lock, but dev deps of package.jsons in the repo do. i think in that case you would need to traverse the in-repo dependency tree and filter things out, similar to the some of the internal logic we have

Reply:
Resolved - not using yarn.lock any more for this